### PR TITLE
[DDW-979] Replace the var substitution logic

### PR DESCRIPTION
**Pull request by cleverca22**
This PR CHANGES.
apply the substituions at the string level, after the `\\` has been parsed to `\`

## Screenshots

- Screenshot

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/messages/GGKFXSKC6)
- [ ] Test

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/main/utils/config.js [hunk 1]
```diff
@@ -3,20 +3,43 @@ import { readFileSync } from 'fs';
 import yamljs from 'yamljs';
 import type { LauncherConfig } from '../config';
 
+function recurseReplace(obj) {
```
Comment (Sam-Jeston) [inline source/main/utils/config.js]: I don't love the implementation of this function. I prefer avoiding for loops in favour of maps, which is certainly possible here. 

But its not a big deal. If this solves the issue, and you get the flow issues resolved we can merge this.

If you have issues I'll assist you when I get online.

**Comment by nikolaglumac** [comment]
@cleverca22 @Sam-Jeston please check my lint/flow fixes.

@alan-mcnicholas please test the Windows installers from the latest build from this PR https://buildkite.com/input-output-hk/daedalus/builds/8648

Once you all approve this PR we can merge it...

File: source/main/utils/config.js (hunk 1 above)
Comment (cleverca22) [reply source/main/utils/config.js]: can always improve it in a future PR when things are less busy

**Comment by nikolaglumac** [comment]
@cleverca22 @disassembler @alan-mcnicholas is it expected that the Selfnode build works on Windows? Because I can see only the "Starting cardano node" screen and there are no cardano-wallet or jormungandr processes.

Screenshot 2019-11-08 at 09 32 06

On all other builds (QA and the Nightly) Node instantly crashes. I guess that could be connected to the current state of the testnets...
The good side is that I can see cardano-wallet.exe and jormungandr.exe are correctly being terminated on Daedalus exit now.

**Comment by nikolaglumac** [comment]
@cleverca22 @alan-mcnicholas so in the selfnode case I see no "cardano-wallet.exe" process and I see that one when I run Dadalus on OSX.
Also, on Daedalus exit I can see that the jormungandr.exe process remains active (all others get terminated) - which means that next time I start Daedalus I get 2 jormungandr.exe processes...

Screenshot 2019-11-08 at 09 46 24
Screenshot 2019-11-08 at 09 47 00

**Comment by DmitriiGaico** [comment]
@cleverca22 @alan-mcnicholas 
Hey! For the QA build i see Cardano Node crashed instantly right after start. Could you please advise if this is expected.
image

**Comment by nikolaglumac** [comment]
@cleverca22 @disassembler @DmitriiGaico @alan-mcnicholas after upgrade to RC7 I can confirm that all 3 builds work (SelfNode, QA and Nightly).

I will merge this PR and we will handle remaining issues in separate tasks:
https://iohk.myjetbrains.com/youtrack/issue/DDW-1059
https://iohk.myjetbrains.com/youtrack/issue/DDW-1060
https://iohk.myjetbrains.com/youtrack/issue/DDW-1061

**Pull request merged**